src/components/ha-dialog-new.ts

```diff
@@ -0,0 +1,94 @@
+import { MdDialog } from "@material/web/dialog/dialog";
+import { css } from "lit";
+import { customElement, property } from "lit/decorators";
+
+/**
+ * Based on the home assistant design: https://design.home-assistant.io/#components/ha-dialogs
+ *
+ */
+@customElement("ha-dialog-new")
+export class HaDialogNew extends MdDialog {
+  /**
+   * When true the dialog will not close when the user presses the esc key or press out of the dialog.
+   */
+  @property({ attribute: "disable-esc-scrim-cancel", type: Boolean })
```

Comment:
It is not just esc, it is also clicking outside the dialog?

Reply:
Yes it is both because the cancel event has this definition: `Dispatched when the dialog has been canceled by clicking on the scrim or pressing Escape.`

Reply:
https://material-web.dev/components/dialog/#events